jupyterhub/spawner.py

```diff
@@ -329,8 +329,12 @@ def clear_state(self):
     
     def user_env(self, env):
         env['USER'] = self.user.name
-        env['HOME'] = pwd.getpwnam(self.user.name).pw_dir
-        env['SHELL'] = pwd.getpwnam(self.user.name).pw_shell
+        home = pwd.getpwnam(self.user.name).pw_dir
+        if home:
+            env['HOME'] = home
+        shell = pwd.getpwnam(self.user.name).pw_shell
+        if shell:
+            env['SHELL'] = shell
```

Comment:
Maybe a comment in the code for future readers, indicating that why setting an empty `SHELL` is problematic.